CentralityiOS/CentralityiOS/ViewControllers/ModifyTaskModalViewController.h

```diff
@@ -17,16 +18,21 @@
 - (void)didEditTask:(TaskObject *)editedTask toFeed:(ModifyTaskModalViewController *)controller;
 @end
 
-@interface ModifyTaskModalViewController : UIViewController <CategoryModalViewControllerDelegate, DueDateModalViewControllerDelegate>
+@interface ModifyTaskModalViewController : UIViewController <CategoryModalViewControllerDelegate, DueDateModalViewControllerDelegate, ShareModalViewControllerDelegate>
 @property (nonatomic, weak) id <ModifyTaskModalViewControllerDelegate> delegate;
 @property (weak, nonatomic) IBOutlet UILabel *modalTitle;
 @property (weak, nonatomic) IBOutlet UIButton *changeCategoryButton;
 @property (weak, nonatomic) IBOutlet UIButton *changeDateButton;
 @property (weak, nonatomic) IBOutlet UIButton *modifyButton;
+@property (weak, nonatomic) IBOutlet UIButton *shareButton;
 @property (weak, nonatomic) IBOutlet UITextField *taskTitleInput;
 @property (weak, nonatomic) IBOutlet UITextView *taskDescInput;
 @property (weak, nonatomic) TaskObject *taskFromFeed;
 @property CategoryObject *taskCategory;
 @property NSDate *taskDueDate;
 @property NSString *modifyMode;
+@property NSMutableArray<PFUser*>*taskSharedOwners;
```

Comment:
Because you are constantly querying and modifying these, I'd suggest switching the `NSMutableArray` in each of the user groups to a `NSMutableDictionary` with a key of `objectId`. This should both simplify the update code and make it more efficient (since you'll be at `O(1)` for lookup, insertion & deletion rather than `O(N)`)